ts/features/itwallet/machine/eid/actors.ts

```diff
@@ -51,6 +55,23 @@ export const createEidIssuanceActorsImplementation = (
 
     // Reset the wallet store to prevent having dirty state before registering a new wallet instance
     store.dispatch(itwLifecycleStoresReset());
+    // Await the integrity preparation before requesting the integrity key tag
+    const isIntegrityServiceReady = await pollForStoreValue({
+      state: store.getState(),
```

Comment:
I am not sure this state is the latest one on subsequent polling cycles: I suspect `getState` returns the latest state only when `pollForStoreValue` is first called, so future invocations of the timeout callback will use a state that is potentially stale. What do you think?

Reply:
Yes, that could be an issue. @mastro993's proposed refactor should be able to solve the problem. Thank you for digging up the problem!